storm-core/src/clj/org/apache/storm/pacemaker/pacemaker_state_factory.clj

```diff
@@ -47,7 +43,7 @@
     (reify
       ClusterState
       ;; Let these pass through to the zk-state. We only want to handle heartbeats.
-      (register [this callback] (.register zk-state callback))
+      (register [this callback] (.register zk-state callback))  ; need update callback, have questions?? callback is IFn here
```

Comment:
If the callback takes a defined set of parameters that are used consistently lets either create or reuse an interface and then wrap the callback function in a reify call to create an instance of the interface for the callback to talk to.

If it does not take any parameters, it just is a function that is triggered than make it a Runnable.  But I think in this case it is the former, not the latter.